### PR TITLE
Add `mapMany(_:)` operator.

### PR DESCRIPTION
**Pull request by jdisho**
Basically doing the same thing as in the `RxSwiftExt` version of [`mapMany`](https://github.com/RxSwiftCommunity/RxSwiftExt/blob/master/Source/RxSwift/mapMany.swift). 

✌️

(Should update the `README`)

### CODE REVIEW

File: Sources/Operators/MapMany.swift [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+public extension Publisher where Output: Collection {
+    /// Projects each element of a publisher collection into a new publisher collection form.
+    ///
+    /// - parameter transform: A transformation function which applies to each element of the source collection.
+    ///
+    /// - returns: A publisher collection whose elements are the result of invoking the transformation function on each element of the source.
+    ///
+    /// An example usage could look as follows:
+        ///
```
Comment (freak4pc) [inline Sources/Operators/MapMany.swift]: Mind aligning the `///` to the same horizontal placed? Looks a tad weird.

File: Sources/Operators/MapMany.swift [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        ///         .mapMany(String.init)
+        ///         .sink(receiveValue: { print($0) })
+        ///
+        ///    intArrayPublisher.send([10, 2, 2, 4, 3, 8])
+        ///
+        ///    // Output: ["10", "2", "2", "4", "3", "8"]
+        ///    ```
+        ///
+    ///
+    func mapMany<T>(_ transform: @escaping (Output.Element) -> T) -> Publishers.Map<Self, [T]> {
```
Comment (freak4pc) [inline Sources/Operators/MapMany.swift]:
```suggestion
    func mapMany<Result>(_ transform: @escaping (Output.Element) -> Result) -> Publishers.Map<Self, [Result]> {
```

Not a huge deal but maybe a bit nicer

**Comment by freak4pc** [comment]
By the way do you think we should also have `tryMapMany`? not sure if that's too interesting

**Comment by jdisho** [comment]
Updated the PR and addressed your points. @freak4pc  

Regarding `tryMapMany`, why not in an other PR?

**Comment by freak4pc** [comment]
Well, because this one adds `mapMany` to mirror `map`, so we should do the same for the `try` one in that case (At least IMO...). Let's leave as is for now ;)

File: Sources/Operators/MapMany.swift [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+    ///         .sink(receiveValue: { print($0) })
+    ///
+    ///    intArrayPublisher.send([10, 2, 2, 4, 3, 8])
+    ///
+    ///    // Output: ["10", "2", "2", "4", "3", "8"]
+    ///    ```
+    ///
+    ///
+    func mapMany<Result>(_ transform: @escaping (Output.Element) -> Result) -> Publishers.Map<Self, [Result]> {
+        return map { $0.map(transform) }
```
Comment (freak4pc) [inline Sources/Operators/MapMany.swift]:
```suggestion
        map { $0.map(transform) }
```

**Pull request merged**